pkg/controller/deployment/deployment_controller.go

```diff
@@ -322,16 +328,25 @@ func (dc *DeploymentController) deleteReplicaSet(obj interface{}) {
 }
 
 // getDeploymentForPod returns the deployment managing the ReplicaSet that manages the given Pod.
-// TODO: Surface that we are ignoring multiple deployments for a given Pod.
+// If there are multiple deployments for a given Pod, only return the oldest one.
 func (dc *DeploymentController) getDeploymentForPod(pod *api.Pod) *extensions.Deployment {
 	rss, err := dc.rsStore.GetPodReplicaSets(pod)
```

Comment:
Since it's about selectors why can't we go straight from a pod to its deployment? Feels like we should have something like dc.dStore.GetDeploymentsForPod.